server.lua

```diff
@@ -429,17 +429,62 @@ end)
 RegisterNetEvent('qb-vehicleshop:server:checkFinance', function()
     local src = source
     local player = QBCore.Functions.GetPlayer(src)
+    local cash = player.PlayerData.money['cash']
+    local bank = player.PlayerData.money['bank']
     local query = 'SELECT * FROM player_vehicles WHERE citizenid = ? AND balance > 0 AND financetime < 1'
     local result = MySQL.query.await(query, {player.PlayerData.citizenid})
     if result[1] then
-        TriggerClientEvent('QBCore:Notify', src, Lang:t('general.paymentduein', {time = Config.PaymentWarning}))
-        Wait(Config.PaymentWarning * 60000)
-        local vehicles = MySQL.query.await(query, {player.PlayerData.citizenid})
-        for _, v in pairs(vehicles) do
-            local plate = v.plate
-            MySQL.query('DELETE FROM player_vehicles WHERE plate = @plate', {['@plate'] = plate})
-            --MySQL.update('UPDATE player_vehicles SET citizenid = ? WHERE plate = ?', {'REPO-'..v.citizenid, plate}) -- Use this if you don't want them to be deleted
-            TriggerClientEvent('QBCore:Notify', src, Lang:t('error.repossessed', {plate = plate}), 'error')
+        if not Config.FinanceAuto then 
+            TriggerClientEvent('QBCore:Notify', src, Lang:t('general.paymentduein', {time = Config.PaymentWarning}))
+            Wait(Config.PaymentWarning * 60000)
+            local vehicles = MySQL.query.await(query, {player.PlayerData.citizenid})
+            for _, v in pairs(vehicles) do
+                local plate = v.plate
+                MySQL.query('DELETE FROM player_vehicles WHERE plate = @plate', {['@plate'] = plate})
+                --MySQL.update('UPDATE player_vehicles SET citizenid = ? WHERE plate = ?', {'REPO-'..v.citizenid, plate}) -- Use this if you don't want them to be deleted
+                TriggerClientEvent('QBCore:Notify', src, Lang:t('error.repossessed', {plate = plate}), 'error')
+            end
+        else
+            -- Automatically take the money out from their account if they have it... If they do not have it, they have certain amount of time until it is repoed...
+            local allPaid = true;
```

Comment:
Move this above `if not Config.FinanceAuto then ... else` and re-use it in the `if not Config.FinanceAuto` section. This will help reduce code duplication (this for loop is defined twice).